backend/core/models/user.py

```diff
@@ -227,21 +227,76 @@ def trusted_users(cls) -> QuerySet["User"]:
     def supertrusted_seed_users(cls) -> QuerySet["User"]:
         return cls.trusted_users().filter(is_staff=True)
 
+    @classmethod
+    def validate_email_unique_with_plus(cls, email: str, username="") -> str:
+        """Raise ValidationError when similar emails are found in the database.
+
+        Keyword arguments:
+        email -- An email that is going to be written in the database.
+        username -- The logged user's username, used to exclude him/herself from
+                    the validation when updating its own email. Empty for
+                    anonymous users.
+
+        Emails considered similar when:
+            - they share the same non-case-sensitive domain ;
+            - they share, in the local part, the same non case-sensitive string
+              before the `+` symbol.
+
+        Examples of emails considered similar:
+            - bob@example.org
+            - bob+@example.org
+            - bob+tournesol@example.org
+            - BOB+tournesol@example.org
+            - bob+hello@example.org
+            - BOB+HELLO@example.org
+            - etc.
+        """
+        email_split = email.rsplit("@", 1)
+
+        # if there is no `@`, do nothing
+        if len(email_split) == 1:
+            return email
+
+        local_part = email_split[0]
+        local_part_split = local_part.split("+")
+
+        users = User.objects.filter(
+            Q(email__iexact=f"{local_part_split[0]}@{email_split[-1]}")
+            | (
+                Q(email__istartswith=f"{local_part_split[0]}+")
+                & Q(email__iendswith=f"@{email_split[-1]}")
+            ),
+        )
+
+        if username:
+            users.exclude(username=username)
```

Comment:
I had missed that. The queryset is not updated here. 
```suggestion
        if username:
            users = users.exclude(username=username)
```

Reply:
I always do the same mistakes.